plugin/src/main/java/org/opensearch/ml/action/tasks/GetTaskTransportAction.java

```diff
@@ -362,4 +371,62 @@ private boolean matchesPattern(Pattern pattern, String input) {
         Matcher matcher = pattern.matcher(input);
         return matcher.find();
     }
+
+    // TODO: move this method to connector utils class
+    private ConnectorAction createConnectorAction(Connector connector) {
+        Optional<ConnectorAction> batchPredictAction = connector.findAction(BATCH_PREDICT.name());
+
+        Map<String, String> headers = batchPredictAction.get().getHeaders();
+
+        String predictEndpoint = batchPredictAction.get().getUrl();
+        Map<String, String> parameters = connector.getParameters() != null
+            ? new HashMap<>(connector.getParameters())
+            : Collections.emptyMap();
+
+        // Apply parameter substitution only if needed
+        if (!parameters.isEmpty()) {
+            StringSubstitutor substitutor = new StringSubstitutor(parameters, "${parameters.", "}");
+            predictEndpoint = substitutor.replace(predictEndpoint);
+        }
+
+        String url = "";
+        String requestBody = "";
+        String method = "GET";
+
+        switch (getEndpointType(predictEndpoint)) {
+            case "sagemaker":
+                url = predictEndpoint.replace("CreateTransformJob", "DescribeTransformJob");
+                requestBody = "{ \"TransformJobName\" : \"${parameters.TransformJobName}\"}";
+                method = "POST";
+                break;
+            case "openai":
+            case "cohere":
+                url = predictEndpoint + "/${parameters.id}";
+                break;
+            case "bedrock":
+                url = predictEndpoint + "/${parameters.processedJobArn}";
+                break;
+        }
+        return ConnectorAction
+            .builder()
+            .actionType(BATCH_PREDICT_STATUS)
+            .method(method)
+            .url(url)
+            .requestBody(requestBody)
+            .headers(headers)
+            .build();
+
+    }
+
+    private String getEndpointType(String url) {
```

Comment:
why do we need same method 2 times?

Reply:
Yes will be moving the common methods to ConnectorUtils class